core/templates/dev/head/pages/exploration_player/PlaythroughService.js

```diff
@@ -214,15 +228,30 @@ oppia.factory('PlaythroughService', [
             !this.isExplorationWhitelisted(playthrough.expId)) {
           return;
         }
-        playthrough.actions.push(LearnerActionObjectFactory.createNew(
-          ACTION_TYPE_EXPLORATION_START,
-          {
-            state_name: {
-              value: initStateName
-            }
-          },
-          CURRENT_ACTION_SCHEMA_VERSION
-        ));
+        // Sometimes, the answer submit action gets recorded before the start.
+        // This happens when an answer is submitted very soon into an
+        // exploration.
+        if (playthrough.actions.length > 0) {
+          playthrough.actions.unshift(LearnerActionObjectFactory.createNew(
```

Comment:
Avoid duplicated code by creating the object outside of these if branches.

Reply:
Done.